echopop/typing/inversion.py

```diff
@@ -0,0 +1,682 @@
+
+        # Update the parameters
+        self.parameters = scaled_params
+
+    def unscale(self):
+        """
+        Convert scaled [0,1] parameters back to their original units and ranges.
+
+        This method reverses the scaling transformation, restoring parameters
+        to their original physical units and bounds. It uses the preserved
+        bounds from `parameter_bounds` to perform the inverse transformation.
+
+        Notes
+        -----
+        The unscaling transformation is:
+
+        .. math::
+            x_{unscaled} = x_{scaled} \\cdot (x_{max} - x_{min}) + x_{min}
+
+        After unscaling, parameters return to their original bounds and units.
+        The scaling status flag is updated to False after this operation.
+
+        Raises
+        ------
+        ValueError
+            If called on parameters that are not currently scaled
+        """
+
+        # Update scaling status
+        self.scaled = False
+
+        # Undo the min-max normalization
+        unscaled_params = {
+            key: {
+                **kwargs,
+                "value": (
+                    kwargs["value"]
+                    * (self.parameter_bounds[key]["max"] - self.parameter_bounds[key]["min"])
+                    + self.parameter_bounds[key]["min"]
+                ),
+                "min": self.parameter_bounds[key]["min"],
+                "max": self.parameter_bounds[key]["max"],
+            }
+            for key, kwargs in self.parameters.items()
+        }
+
+        # Update the parameters
+        self.parameters = unscaled_params
+
+    def to_lmfit(self):
+        """
+        Convert internal parameters to an lmfit.Parameters object.
+
+        This method creates an lmfit.Parameters object compatible with
+        lmfit optimization routines. All parameter attributes (value, min,
+        max, vary) are transferred to the lmfit format.
+
+        Returns
+        -------
+        lmfit.Parameters
+            Parameters object ready for use with lmfit minimization routines.
+            Contains all parameter values, bounds, and vary flags.
+
+        Notes
+        -----
+        The lmfit.Parameters object provides the interface for scipy.optimize
+        and other optimization backends used by lmfit. Default bounds are
+        0.0 (min) and np.inf (max) if not specified.
+
+        Examples
+        --------
+        >>> params = InvParameters({'g': {'value': 1.02, 'min': 1.0, 'max': 1.1}})
+        >>> lmfit_params = params.to_lmfit()
+        >>> # Can now use with lmfit.minimize()
+        """
+        params = Parameters()
+        for name, kwargs in self.parameters.items():
+            params.add(
+                name,
+                value=kwargs["value"],
+                min=kwargs.get("min", 0.0),
+                max=kwargs.get("max", np.inf),
+                vary=kwargs.get("vary", True),
+            )
+        return params
+
+    def unscale_dict(self, scaled_dict: dict) -> dict:
```

Comment:
I think it is confusing to have a `unscale_dict` when there is already `unscale`. The number of use cases for `unscale_dict` is small, so I suggest using `unscale` to perform the operations.

Reply:
I think it is cleaner to have both the scaled and unscaled parameters BOTH as attributes of the InvParameters object. The optimization method can then be set to use either one of them, but it is not something being handled on the InvParameters side.